include/yamail/resource_pool/async/detail/pool_impl.hpp

```diff
@@ -202,15 +312,31 @@ async::stats pool_impl<V, M, I, Q>::stats() const noexcept {
 template <class V, class M, class I, class Q>
 void pool_impl<V, M, I, Q>::recycle(list_iterator res_it) {
     unique_lock lock(_mutex);
-    storage_.recycle(res_it);
-    perform_one_request(lock);
+    io_context_t* io_context = nullptr;
+    queued_handler_t handler;
+    if (!_callbacks->pop(io_context, handler)) {
```

Comment:
> It is better to get a handler as a return value from `pop()` and produce `operator bool` for the `queued_handler` rather than use an out parameter.

I will do this in a separate pr. Such change is quite big because it touches a lot of tests.

Reply:
> And IMO it is better to store `list_iterator_handler` in the `_callbacks` and create an instance of `queued_handler` right after `pop()` instead of changing a state of the object.

Done